tests/bluetooth/audio/cap_commander/src/test_broadcast_reception.c

```diff
@@ -371,3 +450,172 @@ ZTEST_F(cap_commander_test_broadcast_reception, test_commander_reception_start_i
 	zexpect_call_count("bt_cap_commander_cb.broadcast_reception_start", 0,
 			   mock_cap_commander_broadcast_reception_start_cb_fake.call_count);
 }
+
+ZTEST_F(cap_commander_test_broadcast_reception, test_commander_reception_stop_default_subgroups)
+{
+	int err;
+
+	err = bt_cap_commander_register_cb(&mock_cap_commander_cb);
+	zassert_equal(0, err, "Unexpected return value %d", err);
+
+	test_broadcast_reception_start(&fixture->start_param);
+
+	for (size_t i = 0U; i < CONFIG_BT_MAX_CONN; i++) {
+		fixture->stop_param.param[i].src_id = src_id[i];
+	}
+
+	test_broadcast_reception_stop(&fixture->stop_param);
+}
+
+ZTEST_F(cap_commander_test_broadcast_reception, test_commander_reception_stop_one_subgroup)
+{
+	int err;
+
+	err = bt_cap_commander_register_cb(&mock_cap_commander_cb);
+	zassert_equal(0, err, "Unexpected return value %d", err);
+
+	test_broadcast_reception_start(&fixture->start_param);
+
+	/* We test with one subgroup, instead of CONFIG_BT_BAP_BASS_MAX_SUBGROUPS subgroups */
+	for (size_t i = 0U; i < CONFIG_BT_MAX_CONN; i++) {
+		fixture->stop_param.param[i].num_subgroups = 1;
+		fixture->stop_param.param[i].src_id = src_id[i];
+	}
+
+	test_broadcast_reception_stop(&fixture->stop_param);
+}
+
+ZTEST_F(cap_commander_test_broadcast_reception, test_commander_reception_stop_double)
+{
+	int err;
+
+	err = bt_cap_commander_register_cb(&mock_cap_commander_cb);
+	zassert_equal(0, err, "Unexpected return value %d", err);
+
+	test_broadcast_reception_start(&fixture->start_param);
+
+	for (size_t i = 0U; i < CONFIG_BT_MAX_CONN; i++) {
+		printk("Source ID %d: %d %d\n", i, fixture->stop_param.param[i].src_id, src_id[i]);
+
+		fixture->stop_param.param[i].src_id = src_id[i];
+	}
+
+	test_broadcast_reception_stop(&fixture->stop_param);
+
+	test_broadcast_reception_stop(&fixture->stop_param);
+
+	/*
+	 * Since the 2nd reception_stop procedure never completed initiating a new CAP procedure
+	 * should result in an error -EBUSY
+	 * we try with broadcast_reception_start, but could use any CAP procedure for this test
+	 */
+	err = bt_cap_commander_broadcast_reception_start(&fixture->start_param);
+	zassert_equal(-EBUSY, err, "Unexpected return value %d", err);
```

Comment:
Continue from: https://github.com/zephyrproject-rtos/zephyr/pull/76263#discussion_r1715310725

Why doesn't the second `test_broadcast_reception_stop` never complete? It should either complete or return an error. Getting stuck in a busy state seems incorrect

Reply:
The second test_broadcast_reception_stop never completes because we do not get a receive-state callback. However, in the broadcast_reception_stop procedure we can not know if we will never get a callback, since we do not know if someone else started broadcast reception or not, i.e. remembering whether we already did a broadcast reception stop, or not, is not going to fix this issue. We could add something like a time-out, but as per previous discussion this is better left to be handled in the application

Reply:
Since we always do a GATT write with response, we will _always_ (unless the remote server is non-compliant) get a response from our write, and as such there will always be a callback. 

For example if the receive state already has pa_sync_state = 0 and BIS_sync state = 0 (not synced to either), then the mod source operation will likely not trigger a receive state change, but may still provide a write response with `BT_ATT_ERR_SUCCESS`, which should trigger the next step of the stop (rem_src). 

If on the other hand the source is already removed, then we would get a write response with `BT_BAP_BASS_ERR_INVALID_SRC_ID` which we can use to trigger the completed callback with the error. 

Do you have an example where we cannot decidedly complete the CAP procedure?